torch/csrc/jit/codegen/cuda/executor.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <torch/csrc/jit/codegen/cuda/executor_launch_params.h>
+#include <torch/csrc/jit/codegen/cuda/executor_utils.h>
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_cloner.h>
+#include <torch/csrc/jit/codegen/cuda/ir_printer.h>
+#include <torch/csrc/jit/codegen/cuda/kernel.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+struct TORCH_CUDA_API CompileOptions {
+  int device = 0;
+};
+
+class TORCH_CUDA_API FusionExecutor {
+ public:
+  FusionExecutor() {}
+  FusionExecutor(CompileOptions _options) : options_(_options) {}
+
+  std::string KernelName() const {
+    std::stringstream ss;
+    ss << "kernel"; // << fusion_id;
+    return ss.str();
+  }
+
+  static std::string Namespace() {
+    return "CudaCodeGen";
+  }
+
+  std::string getKernel();
+
+  void compileFusion(Fusion* fusion);
+
+  LaunchParams computeLaunchParams(const at::ArrayRef<IValue>& aten_inputs);
+
+  std::vector<at::Tensor> runFusion(
```

Comment:
this may sound minor, but shouldn't this be `runKernel` ? I mean, an executor compiles a fusion -> kernel, then runs the _kernel_?

when I first read the test_gpu.cpp changes it seemed strange to see:
1. compileFusion()
2. runFusion()
.. what exactly are we doing to the poor fusion, compile it or run it??

Reply:
compile implies a transformation: we compile a fusion into a kernel. The names should reflect that (ex. you compile c++ code and run the binary. if someone would say they run C++ code you'd probably think of an interpreter)